android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java

```diff
@@ -516,171 +556,189 @@ public void setIncognito(WebView view, boolean enabled) {
       CookieManager.getInstance().removeAllCookie();
     }
 
-    // Disable caching
-    view.getSettings().setCacheMode(WebSettings.LOAD_NO_CACHE);
-    view.clearHistory();
-    view.clearCache(true);
+    view.ifHasInternalWebView(webView -> {
+      // Disable caching
+      webView.getSettings().setCacheMode(WebSettings.LOAD_NO_CACHE);
+      webView.clearHistory();
+      webView.clearCache(true);
 
-    // No form data or autofill enabled
-    view.clearFormData();
-    view.getSettings().setSavePassword(false);
-    view.getSettings().setSaveFormData(false);
+      // No form data or autofill enabled
+      webView.clearFormData();
+      webView.getSettings().setSavePassword(false);
+      webView.getSettings().setSaveFormData(false);
+    });
   }
 
   @ReactProp(name = "source")
-  public void setSource(WebView view, @Nullable ReadableMap source) {
-    if (source != null) {
-      if (source.hasKey("html")) {
-        String html = source.getString("html");
-        String baseUrl = source.hasKey("baseUrl") ? source.getString("baseUrl") : "";
-        view.loadDataWithBaseURL(baseUrl, html, HTML_MIME_TYPE, HTML_ENCODING, null);
+  public void setSource(RNCWebView view, @Nullable ReadableMap source) {
+    view.ifHasInternalWebView(webView -> {
+
+      // Do not reload reload webview if the source prop has not changed
```

Comment:
nit on "reload reload"

But, less-nitty, this could probably use some extra explanation about how it relates to WebView re-use (the comment here doesn't mention it, although the `if` statement checks for `webViewKey`)... I _assume_ the issue is that otherwise the WebView would reload when attached to a new container?